src/com/mendix/widget/PullToRefresh/handlers/PullToRefresh.ts

```diff
@@ -0,0 +1,215 @@
+import * as domClass from "dojo/dom-class";
+import * as domStyle from "dojo/dom-style";
+
+interface Settings {
+    thresholdDistance: number;
+    maximumDistance: number;
+    reloadDistance: number;
+    mainElement: HTMLElement;
+    triggerElement: HTMLElement;
+    pullToRefreshElement: HTMLElement;
+    classPrefix: string;
+    cssProp: string;
+    iconArrow: string;
+    iconRefreshing: string;
+    pullToRefreshText: string;
+    releaseToRefreshText: string;
+    refreshText: string;
+    refreshTimeout: number;
+    onRefresh: ((value: Function) => Promise<void>) | ((value: Function) => void);
+    resistanceFunction: (value: number) => number;
+}
+
+export class PullToRefresh {
+    private settings: Settings;
+    private pullStart: {
+        screenX: number;
+        screenY: number;
+    };
+    private pullMoveY: number;
+    private distance = 0;
+    private distanceResisted = 0;
+    private state: "pending" | "pulling" | "releasing" | "refreshing" | "scrolling";
+    private setup: boolean;
+    private enable: boolean;
+    private timeout: number;
+    private updating: boolean;
+    private lastDistance = 0;
+
+    constructor(params: { pullToRefreshElement: HTMLElement, mainElement: HTMLElement }) {
+        this.onTouchStart = this.onTouchStart.bind(this);
+        this.onTouchMove = this.onTouchMove.bind(this);
+        this.onTouchEnd = this.onTouchEnd.bind(this);
+        this.onReset = this.onReset.bind(this);
+
+        this.settings = {
+            classPrefix: "pull-to-refresh-",
+            cssProp: "min-height",
+            iconArrow: "&#8675;",
+            iconRefreshing: "&hellip;",
+            mainElement: params.mainElement,
+            maximumDistance: 150,
+            onRefresh: () => { mx.ui.reload(); },
```

Comment:
move to constructor param.
The whole file is a generic pull to refresh class. The `reload` is mx specific